lib/mongodb.js

```diff
@@ -160,14 +160,29 @@ MongoDB.prototype.collection = function (model) {
 /*!
  * Convert the data from database to JSON
  *
- * @param {String} model The model name
+ * @param {String|Object} model The model name or the model class
  * @param {Object} data The data from DB
  */
 MongoDB.prototype.fromDatabase = function (model, data) {
+  var self = this;
+  var props;
+  var settings;
+  var relations = [];
   if (!data) {
     return null;
   }
-  var props = this._models[model].properties;
+  if(typeof model === 'string'){
+    model = this._models[model];
+  }
+  props = model.properties;
+  settings = model.settings;
+  if(settings){
+    for(var k in settings.relations){
+      if(settings.relations[k].property){
+        relations[settings.relations[k].property] = settings.relations[k];
+      }
+    }
+  }
```

Comment:
I had to do this because I have to check if the property delivered by mongodb is defined as that property in the relation, because it is stored as property and not as a relation:

```json
"images":{
      "type": "embedsMany",
      "model": "Image",
      "property": "img",
      "options": {
        "validate": true,
        "forceId": false
      }
```